controllers/opportunity/index.js

```diff
@@ -41,6 +41,25 @@ class opportunityController {
       return res.status(400).send(err);
     }
   }
+
+  //getOpportunities
+  async getOpportunities(req, res) {
+    console.log('Inside controller');
+    const queryObject = { ...req.query };
+
+    try {
+      let fetchedOpportunities = await this.opportunityService.getOpportunities(
+        queryObject
+      );
+
+      return res.status(200).json({
+        data: fetchedOpportunities,
+      });
+    } catch (err) {
+      console.log('ERROR IN getOpportunities OpportunityController', err);
+      return res.status(400).send(err);
+    }
+  }
```

Comment:
do you think sending raw error to front-end would be a good idea?

Reply:
As of now, we shall send responses in a format like the below,

**For successful request**
```
{
   "status": "success",
   "data": {
          [opportuities]
    }
}
```

**For unsuccessful request**
```
{
   "status": "fail",
   "error": {
          [error that is thrown]
    }
}
```

If you have any suggestions regarding this, I'll change. I have no idea of what to send for responses for a standard API.

Reply:
CC: @abdus

Reply:
Any Idea on this ? @abdus @Manvityagi

Reply:
Let me check what standard will we follow and get back to you.

Reply:
@rohithmsr this looks good. you can go ahead with this format. just make sure that you process the error message before sending it to the client. for example, a `duplicate field` error thrown by mongoose would not make much sense to the front-end people. Do you understand what I am trying to say?

Reply:
Hey @abdus , as of now,
 if have just added status and data fields,
 if error, there will be status and error fields as mentioned above,  in which, inside error key, raw error will be its value. Once we implement some more features, i guess it will be easy to identify errors like ValidationError, duplicate error etc. 

Please!! not as of now, I think you understood what I said @abdus

Reply:
If I try to make error in spelling of query param , for example, ?blablabla=blabla, still it returns 0 results, I can write a code to check for the query parameters quickly, but in the future if I want to implement pagination, aliasing etc, the code may break.

Even if I try making errors in URL, it says '/Cannot GET resource/' . This error handling is out of scope for this issue, it needs 
an separate issue

If I try to make error in query param like ?type=blablabla, it returns 0 results which is not an error and it is a successful request.  

Please!! I hope you got it @abdus @Manvityagi 
My suggestion is, we shall take care of error handling in a controller folder after some more features are added.